eeg_flow/tasks/epochs_evoked.py

```diff
@@ -0,0 +1,506 @@
+            for cond in epochs.event_id
+        )
+    return (
+        metadata,
+        fig_rt,
+        behavioral_str,
+        epochs,
+        df_counts,
+        df_drops,
+        fig_drops,
+        evokeds,
+    )
+
+
+def _make_metadata(
+    events: NDArray[+ScalarIntType],
+    events_id: dict[str, int],
+    raw: BaseRaw,
+) -> tuple[pd.DataFrame, NDArray[+ScalarIntType], dict[str, int]]:
+    """Create metadata from events for each epoch.
+
+    For each epoch, it shall include events from the range: [0.0, 1.5] s,
+    i.e. starting with stimulus onset and expanding beyond the end of the epoch.
+    Currently it includes [0.0, 0.999].
+
+    Parameters
+    ----------
+    events : array of shape (n_events, 3)
+    events_id : dict
+    raw : Raw
+
+    Returns
+    -------
+    metadata : DataFrame
+        Metadata of the events.
+    events : array of shape (n_events, 3)
+        The events.
+    event_id : dict
+        Mapping str to int of the events
+    """
+    # MNE auto-generate metadata, which also returns a new events array and an event_id
+    # dictionary. The metadata are created from response-locked events around target,
+    # standard and novel stimuli; and include the events target, standard, novel and
+    # response.
+    metadata, events, event_id = make_metadata_mne(
+        events=events,
+        event_id=events_id,
+        tmin=0.0,
+        tmax=0.999,
+        sfreq=raw.info["sfreq"],
+        row_events=["standard", "target", "novel"],
+    )
+    conditions = [
+        (metadata["event_name"].eq("target"))
+        & (pd.notna(metadata["response"]))
+        & (metadata["response"] < 0.2),
+        (metadata["event_name"].eq("target"))
+        & (pd.notna(metadata["response"]))
+        & (metadata["response"] >= 0.2),
+        (metadata["event_name"].eq("target")) & (pd.isna(metadata["response"])),
+        (metadata["event_name"].eq("standard")) & (pd.notna(metadata["response"])),
+        (metadata["event_name"].eq("standard")) & (pd.isna(metadata["response"])),
+        (metadata["event_name"].eq("novel")) & (pd.notna(metadata["response"])),
+        (metadata["event_name"].eq("novel")) & (pd.isna(metadata["response"])),
+    ]
+    choices = [
+        "FalseAlarms_tooquick",
+        "Hits",
+        "Misses",
+        "FalseAlarms",
+        "CorrectRejections",
+        "FalseAlarms",
+        "CorrectRejections",
+    ]
+    metadata["response_type"] = np.select(conditions, choices, default=0)
+    metadata["response_type"].value_counts()
+    metadata.response_correct = False
+    metadata.loc[
+        (metadata["response_type"] == "CorrectRejections"), "response_correct"
+    ] = True
+    metadata.loc[(metadata["response_type"] == "Hits"), "response_correct"] = True
+    metadata.loc[
+        (metadata["response_type"] == "FalseAlarms"), "response_correct"
+    ] = False
+    metadata.loc[(metadata["response_type"] == "Misses"), "response_correct"] = False
```

Comment:
Heads-up here: this is likely the source of the problem you encountered. c.f. https://stackoverflow.com/a/77530368/8324480

The dtype of the column "response_correct" is `object` instead of `bool`; which does explain why the epoch selection was failing (the `pd.DataFrame.eval`).
The best fix would be to correctly create this `response_correct` column, with a `bool` dtype.